libs/sdk-ui-tests-e2e/cypress/integration/01-sdk-ui-dashboard/shareDashboardToEveryone.spec.ts

```diff
@@ -69,4 +71,195 @@ describe("Share Dashboard To Everyone", { tags: ["post-merge_integrated_tiger"]
             shareDialog.shareItemExistsForUserOrGroup(allUsers, false).cancel();
         });
     });
+
+    describe("Check user permissions", () => {
+        const viewUser = "test-user-view";
+        const shareUser = "test-user-share";
+        const editUser = "test-user-edit";
+        const groupName = "groupTestPermission";
+
+        function removeUsersAndGroups() {
+            for (const user of [viewUser, shareUser, editUser]) {
+                Users.deleteUser(user);
+            }
+            Users.deleteGroup(groupName);
+        }
+
+        function createUsersAndGroups() {
+            Users.createGroup(groupName);
+            const userPermissions = [viewUser, shareUser, editUser].map((user) => {
+                const userType = user.split("-")[2];
+                const permission = userType === "edit" ? "MANAGE" : userType === "share" ? "ANALYZE" : "VIEW";
+                Users.createUser(user, [groupName]);
+                return { user, permission };
+            });
+            WorkspaceAccess.assignUserPermissionToWorkspace(getProjectId(), userPermissions);
+        }
+
+        describe("when sharing dashboard with view permission", () => {
+            before(() => {
+                Users.switchToDefaultUser();
+                Api.setEarlyAccess(getProjectId(), permissionsFeatureFlagEarlyAccess);
+
+                removeUsersAndGroups();
+                createUsersAndGroups();
+                DashboardAccess.assignRulePermissionToDashboard(getProjectId(), Dashboards.KPIs);
```

Comment:
As I understand it, we only have one workspace for all tests. So, if we update data via API, it can make other devs confused about where the data was updated.

Can we move these tests to pre_merge with a mock backend?